core/services/synchronization/telemetry_ingress_batch_client.go

```diff
@@ -102,25 +103,20 @@ func (tc *telemetryIngressBatchClient) Start(ctx context.Context) error {
 
 		serverPubKey := keys.FromHex(tc.serverPubKeyHex)
 
-		conn, err := wsrpc.DialWithContext(ctx, tc.url.String(), wsrpc.WithTransportCreds(clientPrivKey, serverPubKey))
-		if err != nil {
-			return fmt.Errorf("Could not start TelemIngressBatchClient, Dial returned error: %v", err)
-		}
-
 		// Initialize a new wsrpc client caller
 		// This is used to call RPC methods on the server
 		if tc.telemClient == nil { // only preset for tests
+			//conn, err := wsrpc.DialWithContext(ctx, tc.url.String(), wsrpc.WithTransportCreds(clientPrivKey, serverPubKey))
+			conn, err := wsrpc.DialUniWithContext(ctx, tc.lggr, tc.url.String(), clientPrivKey, serverPubKey)
+			if err != nil {
+				return fmt.Errorf("Could not start TelemIngressBatchClient, Dial returned error: %v", err)
+			}
 			tc.telemClient = telemPb.NewTelemClient(conn)
+			tc.close = func() {
+				conn.Close()
+			}
 		}
 
-		tc.wgDone.Add(1)
-		go func() {
```

Comment:
this goroutine was unneeeded from what I can tell. Can just synchronously close in the batch clients close